tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -265,3 +265,25 @@ TEST(SGSparseMatrix, from_dense)
 		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
 			EXPECT_EQ(sparseMatrix(featIndex,vecIndex), denseMatrix(featIndex,vecIndex));
 }
+
+TEST(SGSparseMatrix, transposed_square_matrix)
+{
+	const float64_t sparse_level=0.1;
+	const index_t numberOfFeatures=100;
+	const index_t numberOfVectors=100;
+	const index_t randSeed=0;
+
+	SGSparseMatrix<float64_t> sparseMatrix(numberOfFeatures, numberOfVectors);
+	GenerateMatrix<SGSparseMatrix<float64_t> >(sparse_level, numberOfFeatures, numberOfVectors, randSeed, &sparseMatrix);
+
+	SGSparseMatrix<float64_t> sparseMatrixT=sparseMatrix.get_transposed();
+
+	//check dimension
+	EXPECT_EQ(sparseMatrix.num_features, sparseMatrixT.num_vectors);
+	EXPECT_EQ(sparseMatrix.num_vectors, sparseMatrixT.num_features);
+
+	//check contents
+	for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
```

Comment:
Hey @sudk1896, I still see `CamelCases` in variable naming here. So I think there has been a confusion in understanding what @karlnapf was saying about styling issue. What he meant was, if I am not wrong, to send your new test with the style we expect our code to be in. This patch is for adding a *new test* - so let's make that the purpose of this patch. Style change for *existing code* is another issue. Let's make a different patch for that purpose.

Doesn't make sense to have a patch for *new code addition* in a style that we don't want, only to have the style reverted in a subsequent patch.

Please have a look at our [coding style](https://github.com/shogun-toolbox/shogun/wiki/Code-style) wiki.